Parallel testing for plugins

basiszwo opened the pull request.

basiszwo committed:
Refactor plugin load path helper methods

As we need to make plugin parallel testable we need to know about
loaded plugin paths

basiszwo committed:
Ensure spec and features directories exist

basiszwo committed:
Rake tasks to run plugin specs and cukes in parallel

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://plugin-parallel-testing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Open stage</a> or <a href='http://plugin-parallel-testing.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/teatro.log' target='_blank'>view logs</a>.

basiszwo committed:
Add tasks ro run all test suites in parallel (one by one)

basiszwo committed:
Add custom rake task to run cucumber features in parallel.

The official one provided by parallel_test gem requires some optional
parameter which makes the rake task uncomfortable to use.

basiszwo committed:
Add documentation about using parallel testing tasks

[ci skip]

basiszwo committed:
Rerun failed plugin specs

Some plugin specs in openproject-reporting are failing due to some bad
loading conditions. Running them in isolation solves the problem. So
we just rerun the failing specs in a second run.

basiszwo committed:
Cleanup parallel testing rake tasks

basiszwo committed:
Rerun legacy specs in case some failed
